### PR TITLE
SHA512 HMAC Addition

### PR DESCRIPTION
**Pull request by dccutrig**
Add SHA512 encryption/decryption and associated UTs

### CODE REVIEW

File: include/crypto_config.h [hunk 1]
```diff
@@ -160,7 +160,7 @@
 
 // TC Defines
 #define TC_SH_SIZE 8 /* bits */
-#define TC_SN_SIZE 0
+#define TC_SN_SIZE 4
```
Comment (IbraheemYSaleh) [inline include/crypto_config.h:163]: These defines are only used for max sizes right? Are 4 byte SN's the max allowed per the blue book?

File: include/crypto_structs.h [hunk 2]
```diff
@@ -61,6 +61,7 @@ typedef struct
     crypto_gvcid_t gvcid_tc_blk;
     crypto_gvcid_t gvcid_tm_blk[NUM_GVCID];
     uint8_t lpid;
+    uint8_t key_size;
```
[the remark this answers is not in the document]
Comment (IbraheemYSaleh) [reply include/crypto_structs.h:64]: Will ECS and ACS key sizes ever differ? If yes, this needs to be 2 variable right?

**Comment by IbraheemYSaleh** [comment]
Something about this has been bothering me a bit -- the fact that key_size is now a field of the SA.

There are 3 major pieces of CryptoLib -- the core that orchestrates SDLS-bluebook frame construction, the SADB that manages security associations, and the cryptography interface that manages encryption/decryption functionality. 

The KMC Crypto Service crypto interface has the option of calling the /keyInfo endpoint to retrieve the key length for a given key reference -- it doesn't need the SA to have that info.

The libgcrypt interface accesses the ek_ring directly, and should be able to determine the key lengths from that. 
When the libgcrypt interface is upgraded to be able to access pkcs12 keystores ( https://github.com/nasa/CryptoLib/issues/58 ), it should also be able to get the key lengths from the keystore.

It's redundant and brittle to have the SA store the key_length ... The keystores -- accessible via the cryptography interface -- should be what maintains key metadata. The SA should only ever have key references (ekid/akid)

File: include/crypto_config.h [hunk 3]
```diff
@@ -102,8 +102,8 @@
 
 // Generic Defines
 #define NUM_SA 64
-#define SPI_LEN 2 /* bytes */
-#define KEY_SIZE 32
+#define SPI_LEN 2 /* bytes */ 
+#define KEY_SIZE 64
```
[the remark this answers is not in the document]
Comment (dccutrig) [reply include/crypto_config.h:106]: We have some structs outside of the SA that rely on keysize... so I'm not sure the best approach here.

**Comment by dccutrig** [comment]
This is withdrawn, will be re-visited after IV/ARSN work and started on a branch from _dev_

**Pull request closed without merging**